google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/PubSub.java

```diff
@@ -282,137 +282,7 @@ public static PullOption executorFactory(ExecutorFactory executorFactory) {
-   * unique within the topic.
-   *
-   * <p>Example of asynchronously publishing some messages to a topic.
-   * <pre> {@code
-   * String topicName = "my_topic_name";
-   * Message message1 = Message.of("payload1");
-   * Message message2 = Message.of("payload2");
-   * Future<List<String>> future = pubsub.publishAsync(topicName, message1, message2);
-   * // ...
-   * List<String> messageIds = future.get();
-   * }</pre>
-   *
-   * @param topic the topic where the message is published
-   * @param message the first message to publish
-   * @param messages other messages to publish
-   * @return a {@code Future} for the unique, service-generated ids. Ids are in the same order as
-   *     the messages.
-   */
-  Future<List<String>> publishAsync(String topic, Message message, Message... messages);
-
-  /**
-   * Publishes a number of messages to the provided topic. This method returns a list of
-   * service-generated ids for the published messages. Service-generated ids are guaranteed to be
-   * unique within the topic.
-   *
-   * <p>Example of publishing a list of messages to a topic.
-   * <pre> {@code
-   * String topicName = "my_topic_name";
-   * List<Message> messages = new LinkedList<>();
-   * messages.add(Message.of("payload1"));
-   * messages.add(Message.of("payload2"));
-   * List<String> messageIds = pubsub.publish(topicName, messages);
-   * }</pre>
-   *
-   * @param topic the topic where the message is published
-   * @param messages the messages to publish
-   * @return a list of unique, service-generated, ids. Ids are in the same order as the messages.
-   * @throws PubSubException upon failure, if the topic does not exist or if one of the messages has
-   *     empty payload and no attributes
-   */
-  List<String> publish(String topic, Iterable<Message> messages);
-
-  /**
-   * Sends a request to publish a number of messages to the provided topic. This method returns a
-   * {@code Future} object to consume the result. {@link Future#get()} returns a list of
-   * service-generated ids for the published messages. Service-generated ids are guaranteed to be
-   * unique within the topic.
-   *
-   * <p>Example of asynchronously publishing a list of messages to a topic.
-   * <pre> {@code
-   * String topicName = "my_topic_name";
-   * List<Message> messages = new LinkedList<>();
-   * messages.add(Message.of("payload1"));
-   * messages.add(Message.of("payload2"));
-   * Future<List<String>> future = pubsub.publishAsync(topicName, messages);
-   * // ...
-   * List<String> messageIds = future.get();
-   * }</pre>
-   *
-   * @param topic the topic where the message is published
-   * @param messages the messages to publish
-   * @return a {@code Future} for the unique, service-generated ids. Ids are in the same order as
-   *     the messages
-   */
-  Future<List<String>> publishAsync(String topic, Iterable<Message> messages);
+  Publisher publisher(TopicInfo topic) throws IOException;
```

Comment:
make this `getPublisher` (and use the corresponding pattern for getting the subscriber)